Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -101,34 +99,43 @@ theorem card_fintypeIic : Fintype.card (Set.Iic b) = b + 1 := by
 
 theorem card_fintypeIio : Fintype.card (Set.Iio b) = b := by rw [Fintype.card_ofFinset, card_Iio]
 
--- TODO@Yaël: Generalize all the following lemmas to `SuccOrder`
+@[deprecated "Use `Finset.Icc_succ_left_eq_Ioc`" (since := "2025-02-25")]
 theorem Icc_succ_left : Icc a.succ b = Ioc a b := by
   ext x
   rw [mem_Icc, mem_Ioc, succ_le_iff]
 
+@[deprecated "Use `Finset.Ico_succ_right_eq_Icc`" (since := "2025-02-25")]
 theorem Ico_succ_right : Ico a b.succ = Icc a b := by
   ext x
   rw [mem_Ico, mem_Icc, Nat.lt_succ_iff]
 
+@[deprecated "Use `Finset.Ico_succ_left_eq_Ioo`" (since := "2025-02-25")]
 theorem Ico_succ_left : Ico a.succ b = Ioo a b := by
   ext x
   rw [mem_Ico, mem_Ioo, succ_le_iff]
 
+@[deprecated "Use `Finset.Icc_pred_right_eq_Ico`" (since := "2025-02-25")]
 theorem Icc_pred_right {b : ℕ} (h : 0 < b) : Icc a (b - 1) = Ico a b := by
   ext x
   rw [mem_Icc, mem_Ico, lt_iff_le_pred h]
 
+@[deprecated "Use `Finset.Ico_succ_succ_eq_Ioc`" (since := "2025-02-25")]
 theorem Ico_succ_succ : Ico a.succ b.succ = Ioc a b := by
   ext x
   rw [mem_Ico, mem_Ioc, succ_le_iff, Nat.lt_succ_iff]
 
+-- TODO: Decide whether to generalise the following series of lemmas, or delete.
```

Comment:
These ones shouldn't be deleted, I'd say the TODO should be to generalise them.

Reply:
In fact the TODO that you deleted included generalising these!

Reply:
> In fact the TODO that you deleted included generalising these!

Yes, I changed my mind since 2021 😉

Reply:
I do think they should be deleted, because they are the composition of two more fundamental lemmas each time

Reply:
Except that these ones are currently useful simp lemmas, and the two you're talking about aren't (and probably shouldn't both be) - I'd agree if this one wasn't a simp lemma, but it is, and so the composition argument doesn't really apply.